### DISCUSSION TITLE
[GitLab] Dependency Dashboard description getting cut off

### DISCUSSION THREAD

**Discussion by fgblomqvist** [renovatebot/renovate#9428] [Request Help]
I'm working on a large monorepo with lots of packages, and we've been onboarding all the packages to Renovate in waves. Everything looked fine on the Dependency Dashboard before the latest wave of packages. Now the description is getting cut off somewhere in the "Rate Limit" category. Putting the description in a character counter results in something like 24850 chars.

We're thinking that either the request library (super?) or GitLabs API server is cutting off the body of the request after a certain size, or the GitLab API is cutting off the description after a certain size/char limit (perhaps they counted it as 25k?).

Though, according to the GitLab API, the limit should be a million chars. This is on a self-hosted version so perhaps there are things you need to configure. Figured I'd post here to see if anyone else has seen anything similar? (and yes, we do have 100+ rate-limited MRs at the moment 😄 )

**Comment by rarkins** [comment]
Sounds definitely related to this: https://github.com/renovatebot/renovate/blob/0a68e19d7392f3fbc59ecabd89b66f0751acffee/lib/platform/gitlab/index.ts#L621

**Reply by fgblomqvist** [reply]
Can this be removed now then? Looks like they solved their problem 😄

**Reply by rarkins** [reply]
We need to check how far back they back ported the fix and if we need to add a check for GitLab version. i.e. so that we don't remove it and break people running slightly-older versions of GitLab

**Reply by fgblomqvist** [reply]
At first, it looks like it was released in 13.5:
https://gitlab.com/gitlab-org/gitlab/-/merge_requests/40448#note_433864971

However, looking at the commit/tag history, it looks like it was released in 13.4:
https://gitlab.com/gitlab-org/gitlab/-/commits/v13.4.0-ee/lib/gitlab/quick_actions/extractor.rb

And checking the release log, it was indeed in 13.4.0:
https://about.gitlab.com/releases/2020/09/22/gitlab-13-4-released/

I don't know how to navigate their repo fully, couldn't find branches for older releases (I would have thought they kept each minor version on a separate branch so that they can patch it?), so I can't confirm whether it was backported to 12.x. However, I think it's safe to just put this trimming behind a "< 13.4.0" check. If someone still runs and older version and needs this fixed, they could always do some more digging 🤷

**Reply by fgblomqvist** [reply]
I can submit a PR for this later today 🙂

**Reply by fgblomqvist** [reply]
But on that note, would be nice if Renovate had a firm definition of which versions of each platform it supports, to make maintenance easier. Here's the GitLab doc on their releases: https://docs.gitlab.com/ee/policy/maintenance.html
(also happens to mention that they rarely back-port bug-fixes unless they're severe)
They basically only officially support the last 2-3 monthly (minor version) releases, since they stop back-porting security fixes after that. Perhaps it would be reasonable for Renovate to only support the last 2 major version. E.g. 12 and 13 at the moment, and come May this year, it would be 13 and 14. Or something like that 🤷

**Comment by viceice** [comment]
Normally we only support supported versions, but if community helps and we do not need too complex code we also support older versions
